File: trunk/src/java/org/codehaus/testdox/intellij/TestDoxNonJavaFile.java
package org.codehaus.testdox.intellij;

import javax.swing.Icon;

import com.intellij.openapi.vfs.VirtualFile;

import org.codehaus.testdox.intellij.panel.TestDoxModel;

public class TestDoxNonJavaFile extends TestDoxFile {

    public static final String NO_CLASS_MESSAGE = "No class selected";

    public static final TestElement TEST_ELEMENT = new AbstractTestElement() {
        public String displayString() {
            return NO_CLASS_MESSAGE;
        }

        public Icon icon() {
            return IconHelper.getIcon(IconHelper.NOT_JAVA_ICON);
        }
    };

    public TestDoxNonJavaFile(VirtualFile file) {
        super(file, null, null, null, TestMethod.EMPTY_ARRAY);
    }

    public void updateModel(TestDoxModel model) {
        model.setNotJava();
    }
}

File: trunk/src/scala/org/codehaus/testdox/intellij/TestDoxNonJavaFile.scala
package org.codehaus.testdox.intellij

import com.intellij.openapi.vfs.VirtualFile
import org.codehaus.testdox.intellij.ui.TestDoxTableModel

class TestDoxNonJavaFile(file: VirtualFile) extends TestDoxFile(file, null, null, null, TestMethod.EMPTY_ARRAY) {

  def updateModel(model: TestDoxTableModel) {
    model.setNotJava()
  }
}

object TestDoxNonJavaFile {
  val NO_CLASS_MESSAGE = "No class selected"

  val TEST_ELEMENT: TestElement = new AbstractTestElement() {
    val displayString = NO_CLASS_MESSAGE
    val icon = IconHelper.getIcon(IconHelper.NOT_JAVA_ICON)
  }
}
